pandas/core/indexes/datetimes.py

```diff
@@ -2166,6 +2166,44 @@ def _generate_regular_range(start, end, periods, offset):
     return data
 
 
+def _reraise_overflow_as_oob(endpoint, periods, stride, side='start'):
+    """
```

Comment:
I think it might make sense to :
move this to datetimelike.py
remove the side arg (and just o on passing in)

the reason is we have several other uses of checked_add_with_arr which are catching a scalar add overflow and should be quite similar to this (the other is in Timedelta).

the other uses I saw were for array-like input and they don't raise.

Reply:
The `side` arg is needed if we want the error message to be specific.

All non-test usages of checked_add_with_arr in datetimelike/datetimes/timedeltas.  We may be able to consolidate them in datetimelike.  (though im pretty sure that numeric classes should have overflow checking implemented at some point)

Reply:
my point is you are adding a function here, where if there are only 2 uses, its shorter / simpler just to use a try/except directly

Reply:
my point is you are adding a function here, where if there are only 2 uses, its shorter / simpler just to use a try/except directly

Reply:
... you mean like the original implementation of the PR?